examples/mlperf/dataloader.py

```diff
@@ -155,22 +158,53 @@ def process_batch_bert(data: List[dict]) -> dict[str, Tensor]:
     "next_sentence_labels": Tensor(np.concatenate([s["next_sentence_labels"] for s in data], axis=0), dtype=dtypes.float32),
   }
 
-# For train: Stop when we run through all data
-# For val: Wrap around val dataset and never stop
-#     Reference: https://github.com/mlcommons/training/blob/1c8a098ae3e70962a4f7422c0b0bd35ae639e357/language_model/tensorflow/bert/run_pretraining.py, Line 420
-def batch_load_bert(BS:int, val=False):
-  from extra.datasets.wikipedia import get_wiki_train_files, get_wiki_val_files
-  files = get_wiki_val_files() if val else get_wiki_train_files()
-  blob, end = [], False
-  while files: # As long as there is data, keep going
-    while len(blob) < BS and not end: # Fill blob until there is enough for next step
-      blob.extend(load_bert_file(files.pop(0)))
-      if not files: 
-        if val: files = get_val_files()
-        else: end = True # End of train data - avoid pop on empty file list
-    if len(blob) >= BS: # if last train step does not have enough for a full batch
-      yield process_batch_bert(blob[:BS])
-      blob = blob[BS:]
+def shuffle_parts(file_paths: List[str]) -> List[str]:
+  parts = list(set(map(lambda f: int(Path(f).name.split('_')[0]), file_paths))) # Unique part ids (e.g. 0-499)
+  random.Random().shuffle(parts)
```

Comment:
just random.shuffle should work too, do you need the instance?

Reply:
`random.shuffle` will use the seed we set earlier. This has to be different each run.

Reply:
even if it uses the seed we set earlier, it still generates different shuffle right? (unless we reseed it with the same seed)

it's always preferred to make these explicit and not implicitly recreate the class to hide the issue

Reply:
Well the shuffle only ever happens once at the start so from my understanding, each run would have the same seed. We only ever do one epoch for training

Reply:
we will set different seed for each run, the point is that that seed should be the only random seed